opm/output/data/CellDataContainer.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+
+#include <opm/output/data/CellDataContainer.hpp>
+#include <opm/output/data/Cells.hpp>
+
+
+namespace Opm {
+
+    CellDataContainer::CellDataContainer( std::vector<data::CellData> data ) {
+        for (const auto& elm : data)
+            this->data.push_back( std::move(elm) );
```

Comment:
> It probably will, yes, but it's also crossing module boundaries and I've yet to dig up a proper standards reference for it (I will later tonight probably).

I've yet to find a good standard reference, but it seems so far that the preferred way is indeed to move out of the passed-by-value copy into the local instance. `CDC::CDC( std::vector x ) : local( std::move( x ) ) {}` should be right, in particular because `std::vector` does not have a pass-by-value constructor (only copy- and move constructor).